test/verify/check-login

```diff
@@ -93,6 +95,58 @@ account    required     pam_succeed_if.so user ingroup %s""" % m.get_admin_group
 
         self.allow_restart_journal_messages()
 
+
+    @unittest.skipIf("atomic" not in os.environ.get("TEST_OS", ""), "Resetting expired passwords currently only works with --local-ssh.")
+    def testExpired(self):
+        m = self.machine
+        b = self.browser
+
+        m.execute("sed -i 's/.*ChallengeResponseAuthentication.*/ChallengeResponseAuthentication yes/' /etc/ssh/sshd_config", direct=True)
+        m.execute("( ! systemctl is-active sshd.socket || systemctl stop sshd.socket) && systemctl restart sshd.service", direct=True)
+        m.execute("chage -d 0 admin")
+        m.start_cockpit()
+        b.open("/system")
+
+        b.wait_visible("#login")
+        b.wait_not_visible("#conversation-group")
+        b.wait_visible("#password-group")
+        b.wait_visible("#user-group")
+        b.set_val('#login-user-input', "root")
+        b.set_val('#login-password-input', "foobar")
+        b.click('#login-button')
+
+        b.wait_visible("#conversation-group")
+        b.wait_not_visible("#password-group")
+        b.wait_not_visible("#user-group")
+        b.wait_in_text("#conversation-input-label", "You are required to change your password")
+        b.set_val('#conversation-input', 'foobar')
+        b.click('#login-button')
+
+        b.wait_visible("#conversation-group")
+        b.wait_not_visible("#password-group")
+        b.wait_not_visible("#user-group")
+        b.wait_in_text("#conversation-input-label", "New password")
+        b.set_val('#conversation-input', '123foobar!@#')
+        b.click('#login-button')
+
+        b.wait_visible("#conversation-group")
+        b.wait_not_visible("#password-group")
+        b.wait_not_visible("#user-group")
+        b.wait_in_text("#conversation-input-label", "Retype")
+        b.set_val('#conversation-input', '123foobar!@#')
+        b.click('#login-button')
+
+        with b.wait_timeout(10) as r:
+            b.expect_load()
+        b.wait_present("#content")
+        b.wait_text('#content-user-name', 'Administrator')
+
+        self.allow_journal_messages('cockpit-polkit: user admin reauthorization failed',
```

Comment:
Changing the password on the during the login causes the reauthorization feature to fail as the original password is the one that gets stored and then supplied to polkit. Since we don't know the significance of any subsequent steps, I'm not sure if there is any way around this.